### PR TITLE
[FLINK-6835] Update code style documentation

### PR DESCRIPTION
**Pull request by zentol**
Updates the code style section in regards to checkstyle.

Also addresses FLINK-3814.

cc1
cc2

### CODE REVIEW

File: contribute-code.md [hunk 1]
```diff
@@ -137,22 +137,62 @@ It is also possible to attach a patch to a [JIRA]({{site.FLINK_ISSUES_URL}}) iss
...
-- **No wildcard imports**. Do not use wildcard imports in the core files. They can cause problems when adding to the code and in some cases even during refactoring. Exceptions are the Tuple classes, Tuple-related utilities, and Flink user programs, when importing operators/functions. Tests are a special case of the user programs.
-- **No unused imports**. Remove all unused imports.
-- **Use Guava Checks**. To increase homogeneity, consistently use Guava methods checkNotNull and checkArgument rather than Apache Commons Validate.
-- **No raw generic types**. Do not use raw generic types, unless strictly necessary (sometime necessary for signature matches, arrays).
-- **Supress warnings**. Add annotations to suppress warnings, if they cannot be avoided (such as "unchecked", or "serial").
-- **Comments**. Add comments to your code. What is it doing? Add JavaDocs or inherit them by not adding any comments to the methods. Do not automatically generate comments and avoid unnecessary comments like:
+
+### Javadocs
+- **All public/protected methods and classes must have a Javadoc.**
```
Comment (zentol) [inline contribute-code.md]: package private don't require javadocs; whether that is excluded in "non-private" may be up to interpretation.

**Comment by zentol** [comment]
Where do we reference checkstyle in the project docs? I only found this https://ci.apache.org/projects/flink/flink-docs-release-1.3/internals/ide_setup.html#checkstyle, which IMO we should also move to the website because it isn't relevant for users, which as i understand it is the target audience of the project docs.

File: content/contribute-code.html [hunk 2]
```diff
@@ -331,6 +331,7 @@ <h3 id="imports">Imports</h3>
   <li><strong>No wildcard imports.</strong> They can cause problems when adding to the code and in some cases even during refactoring.</li>
   <li><strong>Import order.</strong> Imports must be ordered alphabetically, grouped into the following blocks, with each block separated by an empty line:
     <ul>
+      <li>&lt;imports from flink packages&gt;</li>
```
Comment (greghogan) [inline content/contribute-code.html]: Would it be clearer to state `imports from org.apache.flink.*`?

Comment (zentol) [inline content/contribute-code.html]: yup, will change it.

**Comment by zentol** [comment]
@greghogan could you take another look?

**Comment by zentol** [comment]
@greghogan With the latest plugin version (5.6.1) the imports are adjusted as well.

**Comment by tzulitai** [comment]
@zentol @greghogan what's the status of this PR?

**Comment by zentol** [comment]
@tzulitai IMO it's good to merge.

**Comment by uce** [comment]
I just got around to trying the new setup with IntelliJ and I think it's a very nice improvement compared to what we had before.

+1 to merge

Follow ups:
- Do we have a page where we give details about how to set up IntelliJ with the checkstyle plugin and importing the code style via checkstyle?
- I think having this page in the code as opposed to the project page would feel more natural (same as to what @greghogan suggested?). Having it as Markdown and linking to it from the project page  would give a nicely formatted page.

**Comment by zentol** [comment]
@uce We have a guide to setup checkstyle in the IDE setup section of the documentation: https://ci.apache.org/projects/flink/flink-docs-release-1.4/internals/ide_setup.html#checkstyle
This guide will also be extended in https://github.com/apache/flink/pull/4086 once https://github.com/apache/flink/pull/4112 is merged, which I want to do after the build profile reorganization.

I think the checkstlye guide (both what we merged here and the setup guide) belong on the project website. The documentation is primarily geared towards Flink users, there's no need for a checkstyle section in there. The contributor guide on the website seems like a good place for it.

**Pull request merged**